plugins/modules/foreman_external_usergroup.py

```diff
@@ -0,0 +1,112 @@
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_external_usergroup
+short_description: Manage Foreman User groups
+description:
+  - Create and delete external user groups in Foreman
+author:
+  - "Kirill Shirinkin (@Fodoj)"
+options:
+  name:
+    description:
+      - Name of the group
+    required: true
+    type: str
+  usergroup:
+    description:
+      - Name of the linked usergroup
+    required: true
+    type: str
+  auth_source:
+    description:
+      - Name of the authentication source to be used for this group
+    required: true
+    type: str
+  state:
+    description:
+      - State of the user
+    default: present
+    choices:
+      - present
+      - absent
+    type: str
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: Create an external user group
+  foreman_external_usergroup:
+    name: test
+    auth_source: "My LDAP server"
+    usergroup: "Internal Usergroup"
+    state: present
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+
+
+def main():
+    module = ForemanEntityAnsibleModule(
+        entity_spec=dict(
+            name=dict(required=True),
+            usergroup=dict(required=True, type='str'),
+            auth_source=dict(required=True, type='entity', flat_name='auth_source_id'),
+        ),
+    )
+
+    entity_dict = module.clean_params()
+
+    module.connect()
+
+    params = {"usergroup_id": entity_dict.pop('usergroup')}
+
+    entity = None
+    changed = False
+
+    # There is no way to find by name via API search, so we need
+    # to iterate over all external user groups of a given usergroup
+    for external_usergroup in module.list_resource("external_usergroups", params=params):
+        if external_usergroup['name'] == entity_dict['name']:
+            entity = external_usergroup
+
+    entity_dict['auth_source'] = module.find_resource_by_name('auth_sources', entity_dict['auth_source'], thin=True)
+
+    if not module.desired_absent and entity and entity['auth_source_ldap']['id'] == entity_dict['auth_source']['id']:
```

Comment:
Ewww. Would this line go away if we'd call the param `auth_source_ldap`? I *think* it would as we'd flatten both old and new entities to only have `auth_source_id` and then the comparison would work. We can still add `aliases=['auth_source']` to the params to make it easier for the users

Reply:
Let me try that, didn't know about aliases option.

Reply:
I don't see how aliases are used in the code: https://github.com/theforeman/foreman-ansible-modules/blob/master/plugins/module_utils/foreman_helper.py#L626. Searching in this helper for aliases in general didnt return promising results. How do I use it? This didn't work:

```
auth_source=dict(required=True, aliases=['auth_source_ldap'],type='entity', flat_name='auth_source_id'),
```

=> same result, doesn't see changes

```
(Pdb) desired_entity
{'name': 'admins', 'auth_source_id': 10}
(Pdb) current_entity
{'id': 9, 'name': 'admins'}
```

Reply:
can you try

```python
auth_source_ldap=dict(required=True, aliases=['auth_source'],type='entity', flat_name='auth_source_id'),
```

Reply:
Just realised what you proposed to do. Fixed now, I just renamed the field to auth_source_ldap. I didn't use an alias, because it's unclear how it will behave if there is some other type of external auth source (neither I am aware which ones would that be).